### PR TITLE
Add line segment struct

### PR DESCRIPTION
**Pull request by jbeenie**

### CODE REVIEW

File: PathKit/Sources/PathKit/Line.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import Foundation
+
+public struct LineSegment {
+    /// The starting point of `self`.
+    public var start: Point
+
+    /// The endpoint of `self`.
+    public var end: Point
+
+    /// The euclidian distance from `start` to `end`.
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Line.swift]:
```suggestion
    /// The Euclidean distance from `start` to `end`.
```

File: PathKit/Sources/PathKit/Point.swift [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        return "Point(\(x), \(y))"
+    }
+}
+
+/// Point Linear interpolation.
+public func lerp(_ lhs: Point, _ rhs: Point, at t: Double) -> Point {
+    return lhs + (rhs - lhs) * t
+}
+
+/// Computes the euclidean distance between two points.
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Point.swift]:
```suggestion
/// Computes the Euclidean distance between two points.
```

File: PathKit/Sources/PathKit/Point.swift [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+public func distance(_ a: Point, _ b: Point) -> Double {
+    return (b - a).length
+}
+
+/// Computes the square of the euclidean distance between two points.
+public func distanceSq(_ a: Point, _ b: Point) -> Double {
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Point.swift]:
```suggestion
public func squareDistance(_ a: Point, _ b: Point) -> Double {
```

File: PathKit/Sources/PathKit/Vector.swift [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        lhs.dy /= rhs
+    }
+
+    /// Returns the vector's magnitude.
+    public var length: Double {
+        return hypot(dx, dy)
+    }
+
+    /// Returns the vector's magnitude squared.
+    public var lengthSquared: Double {
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Vector.swift]:
```suggestion
    public var squaredLength: Double {
```

File: PathKit/Sources/PathKit/Point.swift [hunk 5]
```diff
@@ -46,6 +46,35 @@ extension Point {
         self.x = Double(x)
         self.y = Double(y)
     }
+
+    /// Ceeates a point located at the head of `vector`.
+    ///
+    /// - Parameter vector: The vector used to create the point.
+    public init(_ vector: Vector) {
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Point.swift]: Not a fan of this.

File: PathKit/Sources/PathKit/Vector.swift [hunk 6]
```diff
@@ -50,12 +50,21 @@ extension Vector {
         self.dy = Double(dy)
     }
 
-    /// Creates a vector pointing to `point`.
+    /// Creates a vector pointing from the origin to `point`.
     ///
     /// - Parameter point: The point used to create the vector.
     public init(_ point: Point) {
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Vector.swift]: Not a fan of this either.

File: PathKit/Sources/PathKit/Vector.swift [hunk 7]
```diff
@@ -50,12 +50,21 @@ extension Vector {
...
-        dy = point.y
+        self.init(tail: .zero, head: point)
+    }
+
+    /// Creates a vector pointing from `tail` to `head`.
+    ///
+    /// - Parameters:
+    ///   - tail: The point representing the tail of the vector.
+    ///   - head: The point representing the head of the vector.
+    public init(tail: Point, head: Point) {
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Vector.swift]: Why is tail first?

Comment (jbeenie) [reply PathKit/Sources/PathKit/Vector.swift]: image

Comment (jbeenie) [reply PathKit/Sources/PathKit/Vector.swift]: That was my reasoning

Comment (jbeenie) [reply PathKit/Sources/PathKit/Vector.swift]: Whats the reasoning for putting head first?

Comment (alejandro-isaza) [reply PathKit/Sources/PathKit/Vector.swift]: ok, makes sense

**Review by alejandro-isaza — APPROVED** [review]

File: PathKit/Sources/PathKit/Vector.swift (hunk 6 above)
Comment (jbeenie) [reply PathKit/Sources/PathKit/Vector.swift]: Would you just completely remove this?

File: PathKit/Sources/PathKit/Point.swift (hunk 5 above)
Comment (jbeenie) [reply PathKit/Sources/PathKit/Point.swift]: Would you just completely remove this?

File: PathKit/Sources/PathKit/Vector.swift (hunk 6 above)
Comment (alejandro-isaza) [reply PathKit/Sources/PathKit/Vector.swift]: I would give it a better name. For instance `point.vector(from: .origin)`

File: PathKit/Sources/PathKit/Point.swift (hunk 5 above)
Comment (alejandro-isaza) [reply PathKit/Sources/PathKit/Point.swift]: I would give it a better name. For instance `Point.origin.displace(by: vector)`

Comment (jbeenie) [reply PathKit/Sources/PathKit/Point.swift]: I guess we can just do `.zero + vector`

File: PathKit/Sources/PathKit/Vector.swift (hunk 6 above)
Comment (jbeenie) [reply PathKit/Sources/PathKit/Vector.swift]: Can also just do `Vector(head: point, tail: origin)`

**Pull request merged**